src/Storages/MergeTree/MergeTask.cpp

```diff
@@ -66,7 +66,10 @@ static void extractMergingAndGatheringColumns(
 
     /// Force version column for Replacing mode
     if (merging_params.mode == MergeTreeData::MergingParams::Replacing)
+    {
+        key_columns.emplace(merging_params.is_deleted_column);
```

Comment:
Should this column always be added, or only when it is defined en ENGINE?

Reply:
Not sure to get it; as the version column (called on the next line), it's an optional column. So I applied the same logic.
Can you elaborate ?

Reply:
Nevermind, I see it's optional now. :)